src/control/lib/netdetect/netdetect.go

```diff
@@ -798,20 +821,15 @@ func ValidateProviderConfig(device string, provider string) error {
 		return errors.Errorf("device: %s is an invalid device name", device)
 	}
 
-	var hfiDeviceCount int
-	for _, device := range deviceScanCfg.hwlocDeviceNames {
-		if strings.Contains(device, "hfi1_") {
-			hfiDeviceCount++
-			log.Debugf("There are %d hfi1 devices in the system", hfiDeviceCount)
-		}
-	}
+	hfiDeviceCount := getHFIDeviceCount(deviceScanCfg.hwlocDeviceNames)
+	log.Debugf("There are %d hfi1 devices in the system", hfiDeviceCount)
 
 	// iterate over the libfabric records that match this provider
 	// and look for one that has a matching device name
 	// The device names returned from libfabric may be device names such as "hfi1_0" that maps to a system device "ib0"
 	// or may be a decorated device name such as "hfi1_0-dgram" that maps to "hfi1_0" that maps to a system device "ib0"
 	// In order to find a device and provider match, the libfabric device name must be converted into a system device name.
-	// GetAffinityForDevice() is used to provide this conversion.
+	// GetAffinityForDevice() is used to provide this conversion inside deviceProviderMatch()
```

Comment:
Might be worth putting the meat of this comment next to deviceProviderMatch since that's where the logic is now.